File: JavaScript/ex8_count_numbers.js
// 8. Create a list with random numbers and count every number

const prompt = require ("prompt-sync")();

class CountNumbers{
    counstructor(){
        console.log("This program creates list and counts its numbers")
    }

    create_list(){
        this.list = [];
        this.list_length = parseInt(prompt("How many digits? "), 10);
        for (let i = 0; i < this.list_length; i++){
            this.list.push(Math.floor(Math.random() * this.list_length));
        }
    }

    show_list(){
        console.log("Your list is:", this.list)
    }
    
    create_dictionary(){
        this.dictionary = {};
        for (let i of this.list){
            this.dictionary[i] = 0;
        }
    }

    count_numbers(){
        for (let i of this.list){
            if (this.dictionary[i] === 0){
                this.dictionary[i] = 1;
            }
            else{
                this.dictionary[i] = this.dictionary[i] + 1;
            }
        }
    }

    get_outcome(){
        return this.dictionary;
    }
}

if (require.main === module){
    try {
        let count_numbers = new CountNumbers();
        count_numbers.create_list();
        count_numbers.show_list();
        count_numbers.create_dictionary();
        count_numbers.count_numbers();
        console.log("(Digit, number of digits)", count_numbers.get_outcome())
    }
    catch (error) {
        console.error(error)
    }
}

File: Python/ex8_count_numbers.py
# 8. Create a list with random numbers and count every number

import random
import operator

class CountNumbers:
    def __init__(self):
        print("This program creates list and counts its numbers")

    def create_list(self):
        self.list = []
        self.list_length = int(input("How many digits? "))
        for i in range(self.list_length):
            self.list.append(random.randint(0, self.list_length))
    
    def show_list(self):
        print("Your list is:", self.list)

    def create_dictionary(self):
        self.dictonary = {}
        for i in self.list:
            self.dictonary[i] = 0

    def count_numbers(self):
        for i in self.list:
            if self.dictonary[i] == 0:
                self.dictonary[i] = 1
            else:
                self.dictonary[i] = self.dictonary[i] + 1

    def get_outcome(self):
        self.sorted_dictionary = sorted(self.dictonary.items(), key = operator.itemgetter(0))
        return self.sorted_dictionary

if __name__ == "__main__":
    try:
        count_numbers = CountNumbers()
        count_numbers.create_list()
        count_numbers.create_dictionary()
        count_numbers.count_numbers()
        count_numbers.show_list()
        print("(Digit, number of digits)", count_numbers.get_outcome())
    except Exception as wrong_user:
        print(f"Error!!, program shut down due to {wrong_user}")